branchprotection.tf

```diff
@@ -0,0 +1,67 @@
+resource "github_branch_protection" "branch_protection_api_gateway" {
+  repository     = "ApiGateway"
+  branch         = "master"
+  enforce_admins = true
+
+  required_status_checks {
+    strict   = false
+    contexts = ["build"]
+  }
+
+  required_pull_request_reviews {
+    dismiss_stale_reviews           = true
+    require_code_owner_reviews      = true
+    required_approving_review_count = 1
+  }
+}
+
+resource "github_branch_protection" "branch_protection_session_service" {
+  repository     = "SessionService"
```

Comment:
don't know if `github_repository.repository_session_service.name` would work because it's in a different file but we should try.